runtime/solo/src/lib.rs

```diff
@@ -556,50 +553,229 @@ impl pallet_proxy::Config for Runtime {
 
 impl pallet_insecure_randomness_collective_flip::Config for Runtime {}
 
+pub struct SageEngine;
```

Comment:
As future games might come too, I would suggest moving all instances specific to a game into a separate rust module to keep the runtime.rs module lean. It also makes it easier for new devs to understand what is core logic and what belongs in a game. What do you think?